securedrop/models.py

```diff
@@ -327,11 +327,17 @@ def set_hotp_secret(self, otp_secret):
 
     @property
     def totp(self):
-        return pyotp.TOTP(self.otp_secret)
+        if self.is_totp:
+            return pyotp.TOTP(self.otp_secret)
+        else:
+            raise ValueError('{} is not using TOTP'.format(self))
 
     @property
     def hotp(self):
-        return pyotp.HOTP(self.otp_secret)
+        if not self.is_totp:
+            return pyotp.HOTP(self.otp_secret)
+        else:
+            raise ValueError('{} is not using HOTP'.format(self))
```

Comment:
Just clarifying the "potential vulnerability" language in the PR description - we're not using these properties insecurely anywhere in our production code, that said, this is a solid addition for defense in depth.